### ISSUE TITLE
New algorithm request

### ISSUE THREAD

**Issue by NeoTheCapt** [hashcat/hashcat#3751]
**Is your request based on a publicly known cryptographic schema and where can we find information about?**
Empire CMS
http://www.phome.net/

**Describe alternatives you've considered**
No alternative solution.

**Provide an example hash/database/file and a known correct password**
salt1 := "salt1"
salt2 := "salt2"
password := "123456"
hash = "99bb9e12bdc0f136a4ff844dbe6001ba"

**Explain how the cryptographic schema is implemented**
```
function DoEmpireCMSAdminPassword($password,$salt,$salt2){
	$pw=md5($salt2.'E!m^p-i(r#e.C:M?S'.md5(md5($password).$salt).'d)i.g^o-d'.$salt);
	return $pw;
}
```

**Describe the known limitations of the algorithm**
None

**Where do you typically find the algorithm**
Empire CMS

**Additional context**
None

**Comment by matrix** [comment]
output

```
99bb9e12bdc0f136a4ff844dbe6001ba:salt1:salt2:123456       
                                                          
Session..........: hashcat
Status...........: Cracked
Hash.Mode........: 32300 (Empire CMS - Admin password)
Hash.Target......: 99bb9e12bdc0f136a4ff844dbe6001ba:salt1:salt2
Time.Started.....: Wed May 31 10:34:47 2023, (0 secs)
Time.Estimated...: Wed May 31 10:34:47 2023, (0 secs)
Kernel.Feature...: Pure Kernel
Guess.Mask.......: ?d?d?d?d?d?d [6]
Guess.Queue......: 1/1 (100.00%)
Speed.#1.........: 10544.5 kH/s (4.06ms) @ Accel:64 Loops:100 Thr:32 Vec:1
Recovered........: 1/1 (100.00%) Digests (total), 1/1 (100.00%) Digests (new)
Progress.........: 500000/1000000 (50.00%)
Rejected.........: 0/500000 (0.00%)
Restore.Point....: 0/10000 (0.00%)
Restore.Sub.#1...: Salt:0 Amplifier:0-100 Iteration:0-100
Candidate.Engine.: Device Generator
Candidates.#1....: 120123 -> 688370
Hardware.Mon.SMC.: Fan0: 37%
Hardware.Mon.#1..: Util: 31%
```

benchmark

```
METAL API (Metal 263.9)
=======================
* Device #1: Apple M1, 5408/10922 MB, 8MCU

OpenCL API (OpenCL 1.2 (Aug  8 2022 21:29:55)) - Platform #1 [Apple]
====================================================================
* Device #2: Apple M1, skipped

Benchmark relevant options:
===========================
* --force
* --backend-devices=1
* --backend-devices-virtual=1
* --optimized-kernel-enable

-----------------------------------------------
* Hash-Mode 32300 (Empire CMS - Admin password)
-----------------------------------------------

Speed.#1.........: 66113.2 kH/s (61.61ms) @ Accel:8 Loops:1024 Thr:64 Vec:1
```

these days I will open the PR ;)

**Comment by NeoTheCapt** [comment]
> output
> 
> ```
> 99bb9e12bdc0f136a4ff844dbe6001ba:salt1:salt2:123456       
>                                                           
> Session..........: hashcat
> Status...........: Cracked
> Hash.Mode........: 32300 (Empire CMS - Admin password)
> Hash.Target......: 99bb9e12bdc0f136a4ff844dbe6001ba:salt1:salt2
> Time.Started.....: Wed May 31 10:34:47 2023, (0 secs)
> Time.Estimated...: Wed May 31 10:34:47 2023, (0 secs)
> Kernel.Feature...: Pure Kernel
> Guess.Mask.......: ?d?d?d?d?d?d [6]
> Guess.Queue......: 1/1 (100.00%)
> Speed.#1.........: 10544.5 kH/s (4.06ms) @ Accel:64 Loops:100 Thr:32 Vec:1
> Recovered........: 1/1 (100.00%) Digests (total), 1/1 (100.00%) Digests (new)
> Progress.........: 500000/1000000 (50.00%)
> Rejected.........: 0/500000 (0.00%)
> Restore.Point....: 0/10000 (0.00%)
> Restore.Sub.#1...: Salt:0 Amplifier:0-100 Iteration:0-100
> Candidate.Engine.: Device Generator
> Candidates.#1....: 120123 -> 688370
> Hardware.Mon.SMC.: Fan0: 37%
> Hardware.Mon.#1..: Util: 31%
> ```
> 
> benchmark
> 
> ```
> METAL API (Metal 263.9)
> =======================
> * Device #1: Apple M1, 5408/10922 MB, 8MCU
> 
> OpenCL API (OpenCL 1.2 (Aug  8 2022 21:29:55)) - Platform #1 [Apple]
> ====================================================================
> * Device #2: Apple M1, skipped
> 
> Benchmark relevant options:
> ===========================
> * --force
> * --backend-devices=1
> * --backend-devices-virtual=1
> * --optimized-kernel-enable
> 
> -----------------------------------------------
> * Hash-Mode 32300 (Empire CMS - Admin password)
> -----------------------------------------------
> 
> Speed.#1.........: 66113.2 kH/s (61.61ms) @ Accel:8 Loops:1024 Thr:64 Vec:1
> ```
> 
> these days I will open the PR ;)

Good job my friend. Can't wait to see the new feature.

**Comment by matrix** [comment]
closed by #3753
